src/main/java/org/openstreetmap/atlas/geography/sharding/DynamicTileSharding.java

```diff
@@ -448,6 +447,55 @@ protected SwitchList switches()
                 GEOJSON);
     }
 
+    /**
+     * Calculates and saves the counts for each zoom layer lower than firstZoomLayerToGenerate.
+     *
+     * @param firstZoomLayerToGenerate
+     *            the first zoom layer for which to generate counts
+     * @param countsAtHigherZoom
+     *            HashMap containing counts for all {@link SlippyTile}s in
+     *            firstZoomLayerToGenerate+1
+     * @return a HashMap containing counts for all (@link SlippyTile}s in zoomLayerToGenerate and
+     *         below
+     */
+    private HashMap<SlippyTile, Long> calculateTileCountsForAllZoom(
+            final int firstZoomLayerToGenerate, final HashMap<SlippyTile, Long> counts)
```

Comment:
Nit: Any specific reason to use `HashMap` implementation of `Map` for the return value and input?